src/thor/triplegbuilder.cc

```diff
@@ -328,6 +328,9 @@ void SetShapeAttributes(const AttributesController& controller,
     }
     distance_total_pct = next_total;
     double time = distance / cut_itr->speed; // seconds
+    if (time != time) {                      // avoid NaN
```

Comment:
probably [`std::isnan`](https://en.cppreference.com/w/cpp/numeric/math/isnan) is more readable